Fix mpl import and add Python 3.6 support

jjhelmus opened the pull request.

jjhelmus committed:
Fix import test to use non-gui matplotlib backend

jjhelmus committed:
MNT: Re-rendered with conda-smithy 2.0.0

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jjhelmus committed:
Remove NumPy 1.12 builds

Not all dependencies have been built for NumPy 1.12 yet.
Re-enable once these have been built.

jakirkham commented:
Could add `skip: true  # [np==112]` if you wish to just have NumPy 1.12 skipped.
